### ISSUE TITLE
Create VCPKG port

### ISSUE THREAD

**Issue by fabiencastan** [alicevision/AliceVision#767]
Someone started a PR in this direction:
https://github.com/microsoft/vcpkg/pull/8829

It would be good to finalize it, without hacking/patching the CMakeList.txt but doing the fixes needed directly in the repo.

**Comment by fabiencastan** [comment]
@ChaoJia @JackBoosY Would you be interested in helping us on that?

Currently the build of all the dependencies works fine on windows.
The build of the dependencies works almost fine on linux with the exception of alembic which is still a windows-only port file.

We also need to create the port files for https://github.com/alicevision/popsift and https://github.com/alicevision/CCTag.

**Comment by fabiencastan** [comment]
Regarding alembic there is also a build error, due to `-Werror` in debug:
https://github.com/alembic/alembic/issues/263

**Comment by fabiencastan** [comment]
https://github.com/alicevision/vcpkg/tree/ports/popsift/ports/popsift

**Comment by fabiencastan** [comment]
PR created for Alembic: https://github.com/microsoft/vcpkg/pull/10912

**Comment by JackBoosY** [comment]
We very much look forward to you can propose more PRs to add more ports in vcpkg!

**Comment by fabiencastan** [comment]
CCTag PR: https://github.com/microsoft/vcpkg/pull/16719

**Comment by JackBoosY** [comment]
Will continue to add this to vcpkg.

**Comment by JackBoosY** [comment]
One question, I found the temporary code here:
https://github.com/alicevision/AliceVision/blob/6348b0f346bd8b1efbd1c28e10ebf4dded5df5e0/src/CMakeLists.txt#L301-L308

Why disable dynamic boost here?

I disabled the code block and built successfully.

**Comment by JackBoosY** [comment]
Another question:
We have cmake option `ALICEVISION_USE_MESHSDFILTER` here:
https://github.com/alicevision/AliceVision/blob/6348b0f346bd8b1efbd1c28e10ebf4dded5df5e0/src/CMakeLists.txt#L77

However, the configure system checks whether meshsdfilter folder is exists:
https://github.com/alicevision/AliceVision/blob/6348b0f346bd8b1efbd1c28e10ebf4dded5df5e0/src/CMakeLists.txt#L181-L185

I think the code block should requires `ALICEVISION_USE_MESHSDFILTER` is `ON`.

**Comment by simogasp** [comment]
> I think the code block should requires `ALICEVISION_USE_MESHSDFILTER` is `ON`.

Yes, you are right, that piece of code should be surrounded by an `IF (ALICEVISION_USE_MESHSDFILTER)`

**Comment by JackBoosY** [comment]
@simogasp Can you explain why the first question happens?

**Comment by simogasp** [comment]
> @simogasp Can you explain why the first question happens?

Sorry, I have no idea, does disabling auto-linking via `BOOST_ALL_NO_LIB` mean that dynamic linking is also disabled?

**Comment by JackBoosY** [comment]
Can anyone review my PR https://github.com/microsoft/vcpkg/pull/38034?

**Comment by JackBoosY** [comment]
Can anyone review my PR https://github.com/microsoft/vcpkg/pull/38034?

**Comment by JackBoosY** [comment]
Ping for response.
